cart/cart.go

```diff
@@ -0,0 +1,109 @@
+package cart
+
+import (
+	"checkout_system/catalog"
+	"checkout_system/models"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+)
+
+type ProductCart struct {
+	Products map[string][]models.Product
+}
+
+var AvailableProducts models.Products
+
+func LoadCatalog() {
+	products, err := catalog.LoadPrices()
+	if err != nil {
+		logrus.Error("an error occurred in LoadPrices():", err)
+		return
+	}
+	AvailableProducts = products
+
+	return
+}
+
+func New() *ProductCart {
+	return &ProductCart{Products: make(map[string][]models.Product)}
+}
+
+func (c *ProductCart) Contains(sku string) bool {
+	_, ok := c.Products[sku]
+	if !ok {
+		return false
+	}
+
+	return true
+}
+
+func (c *ProductCart) Get(sku string) ProductCart {
+	products := make(map[string][]models.Product)
```

Comment:
Doesn't make sense to create a map here, cause you might not need it (check below). 
So we can rewrite it like: 
```
product, ok := c.Products[sku]
if !ok {
		return ProductCart{}
}
products := make(map[string][]models.Product)
products[sku] = product
```